backend/src/api/controllers/auth.controller.js

```
@@ -0,0 +1,136 @@
+import { ApiError } from "../../exceptions/api.error.js";
+import bcrypt from "bcrypt";
+import { tokenServices } from "../../services/token.service.js";
+
+function validateName(value) {
+  const name = value.trim();
+  const NAME_PATTERN = /^[\p{L}\s'-]{2,30}$/u;
+
+  if (!name) {
+    return "Name is required";
+  }
+
+  if (name.length < 2 && !NAME_PATTERN.test(name)) {
+    return "Name is not valid";
+  }
+}
+
+function validateEmail(value) {
+  const EMAIL_PATTERN = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!value) return "Email is required";
+  if (!EMAIL_PATTERN.test(value)) return "Email is not valid";
+}
+
+function validatePassword(value) {
+  if (!value) return "Password is required";
+  if (value.length < 6) return "At least 6 characters";
+};
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    name: validateName(name),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  }
+
+  if (errors.name || errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+  await userServices.register(name, email, hashedPassword);
+
+  res.send({ message: 'Ok' });
+}
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+    return;
+  }
+
+  user.activationToken = null;
+  user.save();
+  res.send(user);
+}
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userServices.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  await generateTokens(res, user);
```

The `login` function is missing a crucial check required by the task. Before generating tokens, you must verify if the user has activated their account. You can do this by checking if `user.activationToken` is `null`. If it's not `null`, you should return an error prompting the user to activate their email.